### DISCUSSION TITLE
Example Heartbeat Projects - Spawn/Kill Process vs Activate/Deactivate Machine

### DISCUSSION THREAD

**Discussion by nCubed** [keygen-sh/community#99] [General]
In the heartbeat code examples, the [nodejs](https://github.com/keygen-sh/example-node-process-heartbeats) and [python](https://github.com/keygen-sh/example-python-machine-heartbeats) examples are different. Where the nodejs example is spawning a process and killing a process. While the python example is activating and deactivating a machine. I also noticed the nodejs example does have a function for `deactivateMachine`, however it is never used.

My assumption is the python example is more closely aligned with the recommendations for creating a [floating license](https://keygen.sh/docs/choosing-a-licensing-model/floating-licenses/). Whereas the nodejs example isn't quite doing the same thing. Could you provide some clarification as to the differences between the examples?

For reference, we are currently adding a floating/multi-user license model and have been using the reference article as guidance.

**Comment by ezekg** [comment]
These are different use cases. One is for machine activation (Python), and one is for machine activation and process monitoring (Node). A better comparison would be this Node repo, which is only for machine activation: https://github.com/keygen-sh/example-node-machine-activation

**Comment by nCubed** [comment]
From looking at the examples, I don't think any of them fully satisfy a floating license model with a heartbeat to keep the license active. Reviewing the recommendations (article linked in first post) and these samples, there's enough information to create the scenario we're working on, re: multi-user floating license model. Thanks!
